Index: build.conf
===================================================================
--- build.conf	(revision 1918171)
+++ build.conf	(working copy)
@@ -298,6 +298,7 @@
 libs = libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs_util
 msvc-export = private/svn_fs_fs_private.h ../libsvn_fs_fs/fs_init.h
 msvc-delayload = yes
+msvc-libs = ws2_32.lib
 
 [libsvn_fs_x]
 description = Subversion FSX Repository Filesystem Library
@@ -307,6 +308,7 @@
 libs = libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs_util
 msvc-export = ../libsvn_fs_x/fs_init.h
 msvc-delayload = yes
+msvc-libs = ws2_32.lib
 
 # Low-level grab bag of utilities
 [libsvn_fs_util]
@@ -378,7 +380,7 @@
 libs = aprutil apriconv apr xml zlib apr_memcache
        sqlite magic intl lz4 utf8proc macos-plist macos-keychain
 msvc-libs = kernel32.lib advapi32.lib shfolder.lib ole32.lib
-            crypt32.lib version.lib
+            crypt32.lib version.lib ws2_32.lib
 msvc-export = 
         svn_auth.h svn_base64.h svn_cache_config.h svn_checksum.h svn_cmdline.h
         svn_compat.h svn_config.h svn_ctype.h svn_dirent_uri.h svn_dso.h 
@@ -445,6 +447,7 @@
 path = subversion/svnfsfs
 install = bin
 libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
+msvc-libs = ws2_32.lib
 
 # ----------------------------------------------------------------------------
 #
